File: BOJ_1731/BOJ_1731.java
import java.io.*;

class Main{
        public static void main(String[] args)
                        throws IOException{
                        BufferedReader BR = new BufferedReader(new InputStreamReader(System.in));
                        BufferedWriter BW = new BufferedWriter(new OutputStreamWriter(System.out));
                        int N = Integer.parseInt(BR.readLine());
                        int[] DB = new int[256];
                        for(int i=0; i<N; i++){
                                DB[i] = Integer.parseInt(BR.readLine());
                        }
                        int GAP = DB[1]-DB[0];
                        int MUL = DB[1]/DB[0];
                        int ANS = (DB[2]-DB[1] == DB[1]-DB[0])? DB[N-1]+GAP : DB[N-1]*MUL;
                        BW.write(ANS+"\n");
                        BW.flush();

        }
}

File: BOJ_1731/BOJ_1731.py
import sys
BOX = []
for _ in range(int(sys.stdin.readline())):
    BOX.append(int(sys.stdin.readline()))
if(BOX[1]-BOX[0] == BOX[2]-BOX[1]):
    GAP = BOX[1]-BOX[0]
    print(BOX[-1]+GAP)
elif(BOX[1]//BOX[0] == BOX[2]//BOX[1]):
    GAP = BOX[1]//BOX[0]
    print(BOX[-1]*GAP)
